content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -0,0 +1,214 @@
+
+### Prisma Migrate and <inlineCode>db push</inlineCode>
+
+When you apply changes to your Prisma schema with Prisma Migrate or `db push` with the `foreignKeys` relation mode enabled, foreign keys will be created in your database for all `@relation` attributes in your schema.
+
+## Emulate relations in Prisma with the <inlineCode>prisma</inlineCode> relation mode
+
+The `prisma` relation mode emulates some referential integrity checks and referential actions for each Prisma Client query, using some additional database queries and logic.
+
+The `prisma` relation mode is the default option for the MongoDB connector. It should also be set if you use a relational database that does not support foreign keys. For example, [if you use PlanetScale](https://www.prisma.io/docs/guides/database/using-prisma-with-planetscale#how-to-enable-emulation-of-referential-integrity) you should use the `prisma` relation mode.
+
+<Admonition type="warning">
+  There are significant performance implications to emulating referential
+  integrity in the Prisma client. In cases where the underlying database can
+  handle referential integrity with foreign keys, it is usually the preferred
+  choice.
+</Admonition>
+
+Emulation of relations is only available for Prisma Client queries and does not apply to raw queries.
+
+### Which referential integrity checks are emulated?
+
+When you _update_ a record, Prisma will emulate referential integrity checks. This means that when you update a record with a relation to another record, the related record needs to exist. If the record does not exist, Prisma Client will return an error message.
+
+However, when you _create_ a record, Prisma does not emulate any referential integrity checks. You will be able to create invalid data.
+
+### Which referential actions are emulated?
+
+When you _update_ or _delete_ a record with related records, Prisma will emulate referential actions.
+
+<!-- TODO: list actions with brief description? -->
+
+The following table shows which emulated referential actions are available for each database connector:
+
+| Database    | Cascade | Restrict | NoAction | SetNull | SetDefault |
+| :---------- | :------ | :------- | :------- | :------ | :--------- |
+| PostgreSQL  | **✔️**  | **✔️**   | **✔️**‡  | **✔️**  | **❌**†    |
+| MySQL       | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| SQLite      | **✔️**  | **✔️**   | **✔️**‡  | **✔️**  | **❌**†    |
+| SQL Server  | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| CockroachDB | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| MongoDB     | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+
+- † The `SetDefault` referential action is not supported.
+- ‡ The `NoAction` referential action is not supported for PostgreSQL and SQLite. Instead, use the `Restrict` action.
```

Comment:
See https://prisma-company.slack.com/archives/C02EQD1EWAF/p1667846492813319

Reply:
Done, made note to maybe elaborate later